ethereum/contractBackend.go

```diff
@@ -0,0 +1,97 @@
+package ethereum
+
+import (
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/pkg/errors"
+
+	ethchannel "perun.network/go-perun/backend/ethereum/channel"
+	ethwallet "perun.network/go-perun/backend/ethereum/wallet"
+	"perun.network/go-perun/channel"
+	"perun.network/go-perun/wallet"
+)
+
+// ContractBackend provides ethereum specific contract backend functionality.
+type ContractBackend struct {
+	// cb is the instance of contract backend that will be used for all onchain communications.
+	cb *ethchannel.ContractBackend
+}
+
+// InitCustomContractBackend initializes a contract backend with the pre configured ethereum connection and wallet.
+// Use this function ONLY IN TEST PACKAGES to set a simulated backend and test wallets with weak encryption params.
+func InitCustomContractBackend(cb *ethchannel.ContractBackend) *ContractBackend {
+	return &ContractBackend{cb: cb}
+}
+
+// InitContractBackend initializes a connection to blockchain node and
+// sets up a wallet for funding onchain transactions.
+func InitContractBackend(url, userKeystorePath string, userAddr wallet.Address) (_ *ContractBackend, err error) {
```

Comment:
This function is also testing only?

Reply:
yes, see this [comment](https://github.com/direct-state-transfer/dst-go/pull/50#discussion_r444897627) above.